src/TrackingOptOut.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\CoreStandards\TrackingOptOut.
+ */
+
+namespace Netzstrategen\CoreStandards;
+
+class TrackingOptOut {
+
+  /**
+   * @implements admin_init
+   */
+  public static function admin_init() {
+    static::addSettingsFields();
+  }
+
+  /**
+   * @implements init
+   */
+  public static function init() {
+
+    // Register the shortcode to call the opt-out cookie creation script.
+    add_shortcode('opt-out-link', __CLASS__ . '::registerOptOutShortcode');
+
+    // Enqueue frontend scripts.
+    if (get_option('core_standards_opt_out_active')) {
+      add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
+      // Injects the Google Analytics IDs and the opt-out script in the page head
+      // section before any tracking script.
+      add_action('wp_head', __CLASS__ . '::wp_head', 5);
+      add_action('wp_enqueue_scripts', __CLASS__ . '::enqueueScriptsFirstHead', 1);
+    }
+  }
+
+  /**
+   * Adds Opt-Out settings fields to 'Reading' settings page.
+   */
+  public static function addSettingsFields() {
+    register_setting(
+      'reading',
+      'core_standards_opt_out_ids',
+      'esc_attr'
+    );
+    add_settings_field(
+      'core_standards_opt_out_ids',
+      __('Google Analytics IDs', Plugin::L10N),
+      __CLASS__ . '::optOutGoogleIdsFieldRender',
+      'reading',
+      'default'
+    );
+
+    register_setting(
+      'reading',
+      'core_standards_opt_out_active'
+    );
+    add_settings_field(
+      'core_standards_opt_out_active',
+      __('Activate Opt-Out', Plugin::L10N),
```

Comment:
1. I still don't see why we need two settings to toggle one thing… especially because this is about standards; i.e., we assume that all sites are going to need it, so there's little point in making it optional as our initial assumption is that every site needs it. — It's your call… just pointing out once more that the math doesn't add up for me.

1. If you keep two options instead of one, then the first option must be the checkbox to toggle the whole thing on and off. Children always come after their parents. That's UX 101.